### PR TITLE
keep a ptype attribute when `group_nest()`ing

### PR DESCRIPTION
**Pull request by romainfrancois**
This is to help `tidyr`, i.e.  in this branch: https://github.com/tidyverse/tidyr/pull/511

specifically to uncomment this test: https://github.com/tidyverse/tidyr/blob/dce9a0939dffbfd72eff8cb7f68a147aaa969fe9/tests/testthat/test-nest.R#L48

```r
test_that("nesting works for empty data frames", {
  df <- tibble(x = 1:3, y = c("B", "A", "A"))[0, ]

  out <- nest(df, x)
  expect_equal(names(out), c("y", "data"))

  # if (utils::packageVersion("dplyr") > "0.7.99") {
  #   expect_equal(unnest(out), df)
  # }
  expect_equal(nrow(out), 0L)
  expect_equal(length(out$data), 0L)

  out <- nest(df, x, y)
  expect_equal(length(out$data), 1L)
  expect_equal(out$data[[1L]], df)
})
```

### CODE REVIEW

**Comment by romainfrancois** [comment]
@hadley we had that at some point and then we lost it. This is about `nest()` keeping enough information so that `unnest()`ing retrieves the right columns.

File: src/group_indices.cpp [hunk 1]
```diff
@@ -695,5 +695,8 @@ List group_split_impl(GroupedDataFrame gdf, bool keep, SEXP frame) {
     GroupedDataFrame::strip_groups(out_i);
     out[i] = out_i;
   }
+  if (ptype) {
+    out.attr("ptype") = dataframe_subset(data, IntegerVector(0), NaturalDataFrame::classes(), frame);
```
Comment (hadley) [inline src/group_indices.cpp]: I think this should probably be tibble since we consistently turn ungrouped grouped dfs into tibbles.

Comment (romainfrancois) [reply src/group_indices.cpp]: `out.attr("ptype") = <a tibble with these columns>` ?

Comment (hadley) [reply src/group_indices.cpp]: It just looked like you were giving it class `data.frame` not `tbl_df`?

Comment (romainfrancois) [reply src/group_indices.cpp]: `NaturalDataFrame::classes()` gives: 

 ```
static inline CharacterVector classes() {
    return Rcpp::CharacterVector::create("tbl_df", "tbl", "data.frame");
  }
```

the `dataframe_subset()` call gives a zero row data frame, similar to what `data[integer(),]` would, and we start from a `GroupedDataFrame` so it must be a `tbl_df` if it's originally a `grouped_df`

Comment (hadley) [reply src/group_indices.cpp]: Oh ok, I was confused by that name

**Pull request merged**